File: src/main/Java/boj_step14/boj10816.java
import java.io.*;
import java.util.Arrays;
import java.util.LinkedHashMap;

public class boj10816 {
    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(System.out));
        reader.readLine();
        int[] cardSet = Arrays.stream(reader.readLine().split(" ")).mapToInt(Integer::parseInt).toArray();
        reader.readLine();
        int[] keys = Arrays.stream(reader.readLine().split(" ")).mapToInt(Integer::parseInt).toArray();
        LinkedHashMap<Integer, Integer> cardKeySet = new LinkedHashMap<>();
        for (int key: keys) cardKeySet.put(key, 0);
        for (int card: cardSet) if (cardKeySet.containsKey(card)) cardKeySet.put(card, cardKeySet.getOrDefault(card, 0) + 1);
        for (int key: keys) writer.append(cardKeySet.get(key) + " ");
        writer.close();
    }
}

File: src/main/Kotlin/boj_step14/boj10816.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter

fun main() {
    val r = BufferedReader(InputStreamReader(System.`in`))
    val w = BufferedWriter(OutputStreamWriter(System.out))
    val n = r.readLine().toInt()
    val cardset = r.readLine().split(" ").map { it.toInt() }
    val m = r.readLine().toInt()
    val keys = r.readLine().split(" ").map { it.toInt() }
    var keymap = linkedMapOf<Int, Int>()
    keys.forEach {
        keymap[it] = 0
    }
    cardset.forEach {
        if(keymap.containsKey(it)) keymap[it] = keymap.getValue(it) + 1
    }
    keys.forEach {
        w.write("${keymap[it]} ")
    }
    w.close()
}